engine/table/src/main/java/io/deephaven/engine/table/impl/partitioned/PartitionedTableProxyImpl.java

```diff
@@ -553,11 +573,28 @@ public PartitionedTable.Proxy aggAllBy(AggSpec spec, ColumnName... groupByColumn
     public PartitionedTable.Proxy aggBy(Collection<? extends Aggregation> aggregations, boolean preserveEmpty,
             TableOperations<?, ?> initialGroups, Collection<? extends ColumnName> groupByColumns) {
         if (initialGroups == null) {
-            return basicTransform(true, ct -> ct.aggBy(aggregations, preserveEmpty, null, groupByColumns));
+            return basicTransform(
+                    true,
+                    ct -> ct.aggBy(aggregations, preserveEmpty, null, groupByColumns));
+        }
+        if (initialGroups instanceof Table) {
+            // Force a consistent view of initial groups table to be used for all current and future constituents
+            final Table initialGroupsTable = LivenessScopeStack.computeEnclosed(
+                    () -> ((Table) initialGroups).selectDistinct(groupByColumns).snapshot(),
+                    () -> ((Table) initialGroups).isRefreshing(),
+                    Table::isRefreshing);
```

Comment:
So how could this actually do anything other than return false (definitionally basd on the snapshot()?

Reply:
It could't. I just felt like it was better to go through the motions, especially in case the code is ever copy-pasted.